backend/main/chapters/c09_combining_booleans.py

```diff
@@ -0,0 +1,638 @@
+
+    class FalseAndFalse(VerbatimStep):
+        """
+Finally, try:
+
+__program_indented__
+        """
+        program = "False and False"
+
+    class AndExercise(ExerciseStep):
+        """
+Exercise: rewrite the `is_valid_percentage` function from before, using `and` instead of `or`.
+It should still work exactly as before, accepting a number parameter `x`, and passing the same tests.
+        """
+        hints = """
+Our solution with `or` first determines if `x` is an invalid percentage, else concludes validity. Using `and` will do this in reverse.
+You will have to reverse the `return` statements accordingly.
+You will have to change the comparison operators too.
+        """
+
+        def solution():
+            def is_valid_percentage(_, x: int):
+                if 0 <= x and x <= 100:
+                    return True
+                else:
+                    return False
+            return is_valid_percentage
+
+        tests = {
+            -1: False,
+            0: True,
+            50: True,
+            100: True,
+            101: False,
+        }
+
+    final_text = """
+Awesome! A typical solution looks like:
+
+    def is_valid_percentage(x):
+        if 0 <= x and x <= 100:
+            return True
+        else:
+            return False
+            
+As before, we can simplify this solution to:
+
+    def is_valid_percentage(x):
+        return 0 <= x and x <= 100
+
+In Python,
+
+    if 0 <= x and x <= 100:
+
+can also be written as a short hand:
+
+    if 0 <= x <= 100:
+    
+So the solution can be simplified to its final form:
+
+    def is_valid_percentage(x):
+        return 0 <= x <= 100
+        
+    """
+
+
+class TicTacToeExercise(Page):
+    class TicTacToeWinningRow(ExerciseStep):
+        """
+Exercise: check if a row of tic-tac-toe represents a winner, using `and`.
+If you've never heard of tic-tac-toe, you can read about it
+[here](https://en.wikipedia.org/wiki/Tic-tac-toe) and play a few games against a
+computer opponent [here](https://playtictactoe.org/).
+
+We will use a list of strings "X" and "O" to represent a row of tic-tac-toe:
+
+    row = ["X", "O", "O"]
+
+Write a function `is_winning_row` taking one argument `row` like above (a list of 3 strings),
+which prints `We have a winner!` if `row` is a winning row. Otherwise print `Not a winning row.`
+        """
+        hints = """
+There are only two winning rows: ["X", "X", "X"] and ["O", "O", "O"].
+You will need to use `and`, `if`, `==` and list indexing.
+Check if the three row entries are equal to each other.
+You need to make two separate checks using `==`, then connect them using `and`.
+                """
+
+        def solution():
+            @returns_stdout
+            def is_winning_row(_, row: List[str]):
+                if row[0] == row[1] and row[0] == row[2]:
+                    print("We have a winner!")
+                else:
+                    print("Not a winning row.")
+            return is_winning_row
```

Comment:
```suggestion
        def solution(self):
            def is_winning_row(row: List[str]):
                if row[0] == row[1] and row[0] == row[2]:
                    print("We have a winner!")
                else:
                    print("Not a winning row.")
            return is_winning_row
```

The API changed in #60.

Reply:
Got it.